converse.js

```diff
@@ -634,11 +635,35 @@
             },this), 200));
         };
 
+        this.ping = function (jid, timeout, success, error){
+            if (typeof jid === 'undefined'     || jid == null)     { jid = this.bare_jid; }
+                if (typeof timeout === 'undefined' || timeout == null) { timeout = 45; }
+                if (typeof success === 'undefined' ) { success = null; }
+                if (typeof error === 'undefined' ) { error = null; }
+                if (this.connection) this.connection.ping.ping( jid, success, error, timeout );
+        };
+
+        this.pong = function (ping){
+		    converse.connection.ping.pong(ping);
+	        return true;
+    	};
+
+	    this.registerPongHandler = function (){
+			converse.connection.ping.addPingHandler( this.pong );
+    	};
```

Comment:
I think here we need to advertise that converse.js supports XEP-0199.

Something like this (I think in the `registerPongHandler` method):
```javascript
    converse.connection.disco.addFeature('urn:xmpp:ping');
```